### ISSUE TITLE
Evaluate the compose test-network using Rancher Desktop and Moby / dockerd

### ISSUE THREAD

**Issue by jkneubuh** [hyperledger/fabric-samples#616]
Run a test of the test-network (docker-compose) using a combination of Rancher Desktop and moby/containerd.

By default Rancher will get set up to run with containerd and `nerdctl` / `nerdctl compose`.  There are a couple of edge conditions and differences in how nerdctl compose works that will force us to restructure the test network descriptors: 

- volume mounts with explicit file references to missing files will not mount in the container.

- nerdctl compose does not support "overlay compose descriptors."  This is used in the test network setup of couchdb.

- `ps` and `ls` actions are not supported by `nerdctl`.  This makes the network teardown a bit ... messy. 

Instead of refactoring the compose descriptors to juggle these compatibility issues, configure Rancher to use moby/dockerd behind the scenes and see if this gets the test network up and running with minimal issue.  The steps should be: 

0. Turn "off" Docker Desktop. 
1. Install rancherdesktop. 
2. Switch the Kubernetes Settings from containerd -> dockerd 
3. Start the test network (compose) 

Identify any rough edges with this approach and report back to Discussions #594 on results.

**Comment by jkneubuh** [comment]
cc: @varadgit @denyeart @mbwhite

**Comment by jkneubuh** [comment]
Testing the Recipe from Varad + Shivdeep: 

- Install docker and docker-compose: 
```
brew install docker docker-compose
```

- [Install Rancher Desktop](https://docs.rancherdesktop.io/installation#installing-rancher-desktop-on-macos) and switch Rancher from containerd ->> dockerd(moby) in "Kubernetes Preferences" 


- Clone fabric-samples git: 
```
cd ~/GitHub
git clone git@github.com:hyperledger/fabric-samples.git
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.4.1 1.5.2
```


- Bring up the test network: 
```
export DOCKER_SOCK=/var/run/docker.sock
cd ~GitHub/fabric-samples/test-network
./network.sh up createChannel -ca -c mychannel -s couchdb

You will see the following error
Error: Post "https://localhost:7053/participation/v1/channels": dial tcp 127.0.0.1:7053: connect: connection refused
Channel creation failed

Run the following command:
docker-compose -f  docker/docker-compose-test-net.yaml up -d

Rerun
./network.sh up createChannel -ca -c mychannel -s couchdb
```

- Deploy chaincode: 
```
./network.sh deployCC -ccn basic -ccp ../asset-transfer-basic/chaincode-javascript/ -ccl javascript
```

**Comment by jkneubuh** [comment]
I had *very good* results with @varadgit 's recipe running the compose test-network using Rancher's port/shim of docker-compose.  The one rough edge, as Varad found, is that the compose descriptor is mounting a genesis block file on the volume share: 
```yaml
services:
  orderer.example.com: 
    volumes: 
      - ../system-genesis-block/genesis.block:/var/hyperledger/orderer/orderer.genesis.block
```

When this is mounted by rancher's docker-compose, the volume mount fails and the orderer will refuse to start.  This is the same behavior that we saw running `nerdctl compose`.  I think there's a clue in that when bringing up the compose network with rancher, the output complains about a permissions issue mounting the volume:

```
Creating couchdb1 ... 
Creating orderer.example.com ... error
Creating couchdb0            ... 
Creating couchdb1               ... done
ERROR: for orderer.example.com  Cannot start service orderer.example.com: error while creating mount source path '/Users/jkneubuh/github.com/jkneubuh/fabric-samples/test-network/system-genesis-block/genesis.block': chown /Users/jkneubCreating couchdb0            ... done
Creating peer0.org1.example.com ... done
Creating peer0.org2.example.com ... done
Creating cli                    ... done

ERROR: for orderer.example.com  Cannot start service orderer.example.com: error while creating mount source path '/Users/jkneubuh/github.com/jkneubuh/fabric-samples/test-network/system-genesis-block/genesis.block': chown /Users/jkneubuh/github.com/jkneubuh/fabric-samples/test-network/system-genesis-block/genesis.block: permission denied
ERROR: Encountered errors while bringing up the project.
```

But - the good news here is that the genesis.block file is not actually used in the test network.  The code was converted over to use the orderer's channel participation API, and there's no longer any need to bootstrap from a genesis block when the orderers are launched.    Either this is a permissions issue in how rancher mounts empty / missing files in compose, or how rancher runs as a non-root user, or something else, but it's a moot point.  The fix in this case is to: 

a)  remove the genesis block volume mount  and 
b)  set the DOCKER_SOCK env override.

I give this route (Rancher + dockerd(moby) + docker-compose + test-network) a 👍 !   It's a _viable_ workaround for environments where there is a _requirement_ to use the compose test network without Docker desktop.


One issue I did identify on this route:    Rancher runs the compose test network OK, but it does NOT work as a substitute to run KIND on dockerd.  That said.... Rancher's containerd and k3s support is a huge improvement over KIND for working with a local Kubernetes.

For any os:  run nano network + native binaries. 👍 
For Linux/WSL: run an embedded VM and docker.   👍    (or) 
For Linux/WSL : run Rancher (embedded VM) and docker.  👍 
For OSX : run Docker Desktop ($) + compose test network 👍 or 
For OSX : run Docker Desktop ($) + KIND + kube test network 👍 or 
For OSX : run Rancher + dockerd(moby) container runtime + compose test network.  👍  or 
For OSX : run Rancher + containerd + kube test network. 👍  

Thanks @varadgit and @shivdeep-singh-ibm for the great find on running compose using moby on Rancher.

All around:  👍 👍 👍
